### PR TITLE
[Performance] Enable AscendC index score decode for MiniMax-M3

### PR DESCRIPTION
**Pull request by HaoxinZong**
### What this PR does / why we need it?

This PR ports the missing `MsaIndexScore` KV-splitting optimization from ops-transformer PR 11754 onto the current vLLM Ascend `main` implementation and enables the optimized bundled AscendC operator for MiniMax-M3 index-score decode.

The operator port is intentionally narrow. It retains current mainline hardware-profile dispatch, Catlass integration, and vLLM-specific adapters while adding:

- the `kvChunks` tiling-data path for short-M/long-KV workloads;
- A2/A5 scheduler, wide-score flush, and writeback fixes;
- q4 x 275-block FP16/FP8 regression cases and source-level coverage;
- updated operator documentation.

The A5 decode follow-up removes the communication bottleneck around the faster kernel:

- score the complete replicated index-K block table instead of using TP block splitting and collectives;
- apply `init_blocks` and `local_blocks` in the AscendC epilogue for this full-table path;
- reuse the existing A5 TopK postprocess kernel for final indices and selection counts.

A2/A3 keep the existing TP-sharded decode path. The TP-sharded path keeps candidate forcing in its TopK stage because the operator attributes do not carry a global block offset.

Matched Ascend 950 profiling used MiniMax-M3, TP4 x DP2, decode-only requests, and the maximum TP-rank duration within each synchronous DP step. The complete layer path is measured after `ScatterPaCache` through the final TopK output.

| Input / output | Original A5 Triton | Optimized AscendC | Change | Speedup |
|---|---:|---:|---:|---:|
| 16K / 1K | 22.802 us/layer | 20.579 us/layer | -9.75% | 1.108x |
| 128K / 1K | 57.011 us/layer | 26.735 us/layer | -53.11% | 2.132x |

At 128K, this saves 30.276 us per layer, or about 1.726 ms across the 57 sparse-attention layers in one model step. These are IndexScore layer-path device-time observations, not end-to-end latency claims.

### Does this PR introduce _any_ user-facing change?

Yes. MiniMax-M3 uses the bundled AscendC `MsaIndexScore` implementation for both prefill and decode, including the optimized A5 full-table decode path.

### How was this patch tested?

- rebased onto current `main` at `125924bb2cb5ca3f68368f251c01c705d3a67b10`
- stable patch-id verified for the operator-sync commit after rebase
- `python -m py_compile` for the modified model, NPU helper, and unit-test files
- `git diff --check`
- release-environment MiniMax-M3 unit test: 64 passed
- Ascend 950 `MsaIndexScore` package build for the operator-sync commit
- A5 service profiling with two 16K/1K and two 128K/1K requests; all requests returned HTTP 200 with exact prompt/completion token counts, and all eight DP2 x TP4 rank traces were generated

The latest-main three-file test overlay passed the new IndexScore cases; six unrelated cases failed because the test container retained release versions of the hardware-capability enum, scatter API, and C++ source layout. Full main-environment validation is delegated to CI.

- vLLM main: https://github.com/vllm-project/vllm/commit/84030bbe3d74d99bad477a3d2e37a973ccd8865c

### CODE REVIEW

**Comment by yingyingzizi** [comment]
/rerun
[Bot]: rerun command failed: you do not have permission. Only the PR author or users with triage+ permission can trigger /rerun.

File: vllm_ascend/models/minimax_m3/msa_m3.py [hunk 1]
```diff
@@ -507,7 +513,26 @@ def forward(
             tp_group = get_tp_group()
             decode_iq = iq[:num_decode_tokens]
             if _USE_ASCENDC_INDEX_SCORE_DECODE:
-                if tp_group.world_size > 1 and index_md.num_prefills == 0:
+                if get_ascend_device_type() == AscendDeviceType.A5:
```
Comment (kunpengW-code) [inline vllm_ascend/models/minimax_m3/msa_m3.py]: Distinguish device types via `hardware_profile`.

File: tests/e2e/pull_request/one_card/test_msa_index_score.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 import torch
 import torch_npu  # noqa: F401
 
+from vllm_ascend.device.hardware_profile import HardwareCapability, get_current_hardware_profile
```
Comment (ZT-AIA) [inline tests/e2e/pull_request/one_card/test_msa_index_score.py]: https://github.com/vllm-project/vllm-ascend/tree/main/tests/e2e/nightly/single_node

**Review by ZT-AIA — DISMISSED** [review]

**Review by kunpengW-code — DISMISSED** [review]

File: vllm_ascend/models/minimax_m3/msa_m3.py (hunk 1 above)
Comment (yingyingzizi) [reply vllm_ascend/models/minimax_m3/msa_m3.py]: Fixed. Replaced device-type checks with hardware_profile capability checks.

File: tests/e2e/pull_request/one_card/test_msa_index_score.py (hunk 2 above)
Comment (yingyingzizi) [reply tests/e2e/pull_request/one_card/test_msa_index_score.py]: Fixed. Added operator accuracy coverage under tests/e2e/nightly/single_node/ops/singlecard_ops/test_msa_index_score.py and reverted the changes to the PR one-card test.

**Comment by HaoxinZong** [comment]
/rerun
[Bot]: rerun completed. No failed jobs found.

**Comment by CXY-Katrina** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

File: tests/e2e/nightly/single_node/ops/singlecard_ops/test_msa_index_score.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+BLOCK_SIZE = 128
+HEAD_DIM = 128
+NUM_HEADS = 8
+SCORE_ALIGNMENT = 16
+MASK_SIZE = 2048
+FILL_THRESHOLD = -1.0e30
+BOOST_THRESHOLD = 1.0e28
+
+
+def _reference_scores(query, key, block_table, q_lens, kv_lens, sparse_mode, init_blocks, local_blocks):
```
Comment (kunpengW-code) [inline tests/e2e/nightly/single_node/ops/singlecard_ops/test_msa_index_score.py:21]: Do we need to modify any other files when adding this Nightly test case? This should be verified before merging.

Comment (yingyingzizi) [reply tests/e2e/nightly/single_node/ops/singlecard_ops/test_msa_index_score.py:21]: Fixed the missing registration in 904602d. Added msa-index-score-a3 and msa-index-score-a5 to nightly_config.yaml, and forwarded matrix.test_config.tests in the A5 workflow. YAML validation, dispatch resolution (each target and both together), and actionlint passed locally. NPU nightly execution is still pending. Please trigger `/nightly msa-index-score-a3` for pre-merge validation. One A5 prerequisite: /nightly dispatches the workflow from main, whose A5 caller currently does not forward `tests`; that one-line workflow fix must reach main before `/nightly msa-index-score-a5` can execute this pytest case.

**Comment by CXY-Katrina** [comment]
/nightly msa-index-score-a3
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/35439268977)

**Comment by kunpengW-code** [comment]
/nightly msa-index-score-a5
nightly command triggered.
- Nightly-A5: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/35441763340)

**Review by kunpengW-code — APPROVED** [review]

**Comment by kunpengW-code** [comment]
/nightly msa-index-score-a5
nightly command triggered.
- Nightly-A5: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/35481744301)

**Comment by kunpengW-code** [comment]
/nightly msa-index-score-a5
nightly command triggered.
- Nightly-A5: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/35486679463)
